docs/architecture/design-proposals/dashboard/application-view-revamp.md

```diff
@@ -0,0 +1,112 @@
+
+The raw information included in the file is this:
+
+```
+dependencies:
+- name: mariadb
+  version: 7.x.x
+  repository: https://kubernetes-charts.storage.googleapis.com/
+  condition: mariadb.enabled
+  tags:
+    - wordpress-database
+```
+
+From that information, if the user has permissions to request AppRepositories, we should be able to map https://kubernetes-charts.storage.googleapis.com/ to `stable` (name given in the AppRepository) and generate a link for the MariaDB chart:
+
+<img src="./img/chart-dependencies.png" width="300px">
+
+### Improve Application Status Report
+
+Right now we only show a "Ready" or "Not Ready" status message depending on the Application workloads. We take into account deployments, statefulsets and daemonsets. Only if all of those are ready, we show the "Ready" status. Since we already have that information, we can show it to the user in order to identify what can go wrong or what it's missing:
+
+<img src="./img/pod-count.png" width="300px">
+
+### Highlight Application Credentials
+
+In the current view, application credentials can be found either reading and executing the commands that the Notes suggest, or clicking on the icon to reveal the secrets below. Those secrets usually are of the type "Opaque" (rather than `kubernetes.io/tls` or `kubernetes.io/service-account-token`). We can extract those credentials and show them in a more prominent place. That way the user can easily discover the application credentials without the need of a terminal:
+
+<img src="./img/credentials-box.png" width="400px">
+
+Note that we will still be showing the different application Secrets along with all the other resources in its own table.
+
+### Compress Application Actions
+
+The current approach is to show one button per action (Upgrade, Delete, Rollback). This list is growing over time (for example, the backend endpoint for running tests is ready to be used). It can also be confusing to show a different list of buttons depending on the release state. One example is the Rollback button that is only rendered if the application has been upgraded. In order to avoid these issues, we can show a clickable menu with the different options, graying out the options not available (potentially showing a tooltip):
+
+<img src="./img/configuration-options.png" width="300px">
+
+### Include Additional Information for Access URLs
+
+The current list of URLs can be improved with two small changes:
+
+- If the browser can access the URL, we could show an icon so the user knows if the URL is working.
+- If the URL is not working and the cause is well known, we could show additional information for the user to debug it (e.g. Pending IPs when using LoadBalancers in Minikube [link](https://github.com/kubeapps/kubeapps/issues/953)).
```

Comment:
This also sounds very useful. Worth getting input on the UI when Angel or others have time.